packages/expo-doctor/src/checks/StoreCompatibilityCheck.ts

```diff
@@ -0,0 +1,66 @@
+import fs from 'fs';
+import path from 'path';
+import semver from 'semver';
+
+import { DoctorCheck, DoctorCheckParams, DoctorCheckResult } from './checks.types';
+import { learnMore } from '../utils/TerminalLink';
+import { existsAndIsNotIgnoredAsync } from '../utils/files';
+
+export class StoreCompatibilityCheck implements DoctorCheck {
+  description = 'Check if the project meets version requirements for submission to app stores';
+
+  sdkVersionRange = '*';
+
+  async runAsync({ exp, projectRoot }: DoctorCheckParams): Promise<DoctorCheckResult> {
+    let issue: string | undefined = undefined;
+
+    // *** non-CNG ***
+    if (await existsAndIsNotIgnoredAsync(path.join(projectRoot, 'android', 'build.gradle'))) {
+      const buildGradle = fs.readFileSync(
+        path.join(projectRoot, 'android', 'build.gradle'),
+        'utf8'
+      );
+      // checks for exact string from SDK 49 bare template, will not work if it was changed
+      const targetSdkVersionRegex =
+        /^\s*targetSdkVersion\s*=\s*(?:Integer\.parseInt\(findProperty\('android\.targetSdkVersion'\)\s*\?:\s*'(\d+)'\)|'(\d+)')/m;
+      const match = buildGradle.match(targetSdkVersionRegex);
+      const targetSdkVersion = match ? parseInt(match[1], 10) : undefined;
+      if (targetSdkVersion && targetSdkVersion < 34) {
```

Comment:
Might be nice to have the `34` listed out as constant somewhere on the top. That would be easier to update or bump.

In the future, we could also tie in "minimum versions" and their date they kick into effect. Google usually announces this ahead of time, we could incorporate that ahead of time too.